Standardize templates across examples

rhcarvalho opened the pull request:
Changing Django, Rails and Node.js examples.
Tested all examples/templates with creating a project for each, deploying and curl'ing.

@bparees PTAL

rhcarvalho commented:
@ewolinetz @sspeiche PTAL

sspeiche commented:
@rhcarvalho I made a number of these changes already, probably a good idea to rebase. Please I believe @bparees wants to avoid ImageStream and instead use ImageStreamTags

sspeiche commented on openshift/templates/nodejs.json:
```diff
@@ -171,21 +185,17 @@
       "value": "https://github.com/openshift/nodejs-ex.git"
     },
     {
-      "name": "FRONTEND_SERVICE_NAME",
```
What is the service name stuff removed? I think just missed due to timing of pr #5

bparees replied:
yeah I think there's a reversion in this pr.

rhcarvalho commented:
@sspeiche I rebased. Sorry, I've actually moved away from parameterizing the FRONTEND_SERVICE_NAME. That's something I introduced earlier, but looking at all templates it seemed that it was not bringing that much value for the level of indirection.
Do you find it useful?

I'm fine with either ImageStream or ImageStreamTags, let's her from @bparees then and I can update the PRs.

sspeiche commented:
@rhcarvalho I'm fine with getting rid of service name stuff...it just adds to the UI and not sure getting much value out of it

bparees commented:
ok, hardcoding the service name is fine.

but we should be using imagestreamtags not imagestreams.

rhcarvalho commented:
I can change that tomorrow. No computer here now.
The ImageStream references are all converted to ImageStreamTags by appending the ":latest" suffix, what's the reasoning to prefer one over the other in templates?

bparees commented:
@rhcarvalho ImageStream is basically deprecated.  ImageStreamTag is cleaner, since ImageStream required a separate field to indicate the Tag.

bparees commented:
@rhcarvalho also the imagestreamtags we define now have versions..you should use the version, not latest:

https://github.com/openshift/origin/blob/master/examples/image-streams/image-streams-centos7.json#L20

rhcarvalho commented:
I'll move all examples to use ImageStreamTags tomorrow morning, and also add a DB template for node.js (green flag from you @sspeiche?).

@bparees, regarding versions, that makes sense for the image streams we provide in origin, the ones you linked to. But I didn't understand how that applies to the framework examples. So far we're defining ImageStreams in those templates for a out-of-box experience -- independent of some admin adding proper image streams to projects a priori.
Are you suggesting we change the IS definitions to match those in origin?

bparees commented:
@rhcarvalho We're actually going to be removing the imagestreams that the templates define and rely entirely on an admin creating the imagestreams in the openshift namespace.  So the templates will no longer be "self contained".  i'll forward you the email thread.

rhcarvalho commented:
Ok, ok... how about the image streams for application code, the ones we push builds to?

bparees commented:
@rhcarvalho yes we'll still define an output stream.

rhcarvalho commented:
@bparees I've updated rails, django and nodejs examples to use the image streams in the "openshift" namespace and use the version tags (`ruby:2.0`, `python:3.3`, `nodejs:0.10`)

mfojtik commented:
@bparees @rhcarvalho would be nice to create an test in jenkins that start openshift and submit all these templates and check if the frontends are running...

bparees commented:
@mfojtik yes it would :)

why not add it to extended tests?

bparees commented on openshift/templates/nodejs.json:
```diff
@@ -162,21 +173,17 @@
-  "labels": {
-    "template": "nodejs-example"
-  }
+      "value": "nodejs.app"
```
make this look more like a hostname.

rhcarvalho replied:
Please let's do this in a another group of PRs.

mfojtik commented on openshift/templates/nodejs.json:
```diff
@@ -4,17 +4,23 @@
-      "tags": "instant-app,nodejs"
+      "description": "An example Node.js application with no database",
+      "tags": "instant-app,nodejs",
+      "iconClass": "icon-nodejs"
```
make sure we have this icon ;-)

rhcarvalho replied:
@mfojtik we do: https://github.com/openshift/origin/blob/master/assets/app/styles/_openshift-logos-icon.less#L64

mfojtik commented on openshift/templates/nodejs.json:
```diff
@@ -55,67 +64,65 @@
+          {
+            "type": "GitHub",
+            "github": {
+              "secret": "secret123"
```
make this parameter

mfojtik replied:
please make it so in every template. this is very important as the parameters are only way users can change this value when creating applications from UI

rhcarvalho replied:
done

mfojtik commented on openshift/templates/nodejs.json:
```diff
@@ -55,67 +64,65 @@
+            }
+          },
+          {
+            "type": "Generic",
```
i'm not sure if generic hooks have sense here since the repository is github

bparees replied:
agreed, let's remove it.

rhcarvalho replied:
removed from all templates

bparees commented:
lgtm.
